### PR TITLE
CKE4 -> CKE5 migration guide redesign

### PR DESCRIPTION
**Pull request by AnnaTomanek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Type: CKEditor 4 migration guide redesign. Closes cksource/ckeditor5-internal#2752.

---

### Additional information

_The migration guide will be split into an entire CKE4 migration section._

### CODE REVIEW

File: docs/updating/migration-from-ckeditor-4.md [hunk 1]
```diff
@@ -1,735 +1,110 @@
...
 </info-box>
 
-When compared to its predecessor, CKEditor 5 should be considered **a totally new editor**. Every single aspect of it was redesigned &mdash; from installation, to integration, to features, to its data model, and finally to its API. Therefore, moving applications using a previous CKEditor version to version 5 cannot be simply called an "upgrade". It is something bigger, so the "migration" term fits better.
+When compared to its predecessor, CKEditor 5 should be considered **a totally new editor**. Every single aspect of it was redesigned &mdash; from installation, to integration, to features, to its data model, and finally to its API.
 
-There is no "drop in" solution for migrating. In this guide we hope to summarize the most important aspects you need to consider before you proceed with installing CKEditor 5.
+There is no automatic solution for migrating. This section summarizes the most important aspects you need to consider before you proceed with installing CKEditor 5.
```
Comment (apadol-cksource) [inline docs/updating/migration-from-ckeditor-4.md:22]:
```suggestion
There is no automatic solution for migrating. This section summarizes the most important aspects you need to consider before you proceed with moving to CKEditor 5.
```

File: docs/updating/migration-from-ckeditor-4.md [hunk 2]
```diff
@@ -1,735 +1,110 @@
...
+## Differences between CKEditor 4 and CKEditor 5
 
-## Installation and integration
+Here are the key differences between the two editor versions:
 
-The very first aspect that changed with CKEditor 5 is its installation procedure. It became much more modern with the introduction of modular patterns, UMD, npm, etc. Refer to {@link installation/getting-started/quick-start Installation} for more details.
+* CKEditor 5 is written in TypeScript/ES6.
```
Comment (apadol-cksource) [inline docs/updating/migration-from-ckeditor-4.md:27]: Could we change it into the table showing differences side by side?

File: docs/updating/ckeditor4-plugin-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# Scope:
+# Compare CKEditor 4 plugins with their CKEditor 5 equivalents.
+
+category: ckeditor4-migration
+menu-title: Plugin compatibility
+order: 40
+modified_at: 2023-03-17
+---
+
+# CKEditor 4 plugin compatibility
```
Comment (apadol-cksource) [inline docs/updating/ckeditor4-plugin-compatibility.md:11]:
```suggestion
# CKEditor 4 plugin equivalents
```

Plugins are not compatible so it is misleading.

**Review by apadol-cksource — CHANGES_REQUESTED** [review]

File: docs/updating/migration-from-ckeditor-4.md (hunk 1 above)
Comment (godai78) [reply docs/updating/migration-from-ckeditor-4.md:22]: I don't think all of that can be done as a comparison table. CKE5 has some stuff whereas CKE4 has none, so nothing compares.

**Comment by AnnaTomanek** [comment]
Note to self: some new configuration options that need to be added to the list:

*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-applicationTitle](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-applicationTitle)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-clipboard_handleImages](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-clipboard_handleImages)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-colorButton_contentsCss](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-colorButton_contentsCss)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-coreStyles_toggleSubSup](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-coreStyles_toggleSubSup)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached_callback](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached_callback)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached_interval](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-delayIfDetached_interval)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-editorplaceholder_delay](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-editorplaceholder_delay)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-emoji_followingSpace](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-emoji_followingSpace)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-filebrowserImageUploadUrl](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-filebrowserImageUploadUrl) 
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-image2_defaultLockRatio](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-image2_defaultLockRatio)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-maximize_historyIntegration](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-maximize_historyIntegration)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-observableParent](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-observableParent)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-shiftLineBreaks](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-shiftLineBreaks)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-tabletools_scopedHeaders](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-tabletools_scopedHeaders)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-widget_keystrokeInsertLineAfter](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-widget_keystrokeInsertLineAfter)
*   [https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-widget_keystrokeInsertLineBefore](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-widget_keystrokeInsertLineBefore)

CKEditor 4 is a surprisingly active project when it comes to adding new configuration options :see\_no\_evil:

**Pull request merged**